src/utils/form.tsx

```diff
@@ -324,15 +324,12 @@ function defaultSyncHiddenFormInput(
   input.tabIndex = -1;
 
   if (isCheckable(component)) {
+    input.checked = component.checked;
+
     // keep in sync to prevent losing reset value
     input.defaultChecked = component.defaultChecked;
     // heuristic to support default/on mode from https://html.spec.whatwg.org/multipage/input.html#dom-input-value-default-on
     input.value = component.checked ? value || "on" : "";
-
-    // we disable the component when not checked to avoid having its value submitted
-    if (!disabled && !component.checked) {
-      input.disabled = true;
-    }
```

Comment:
For my own understanding, why don't we need this anymore?

Reply:
Good question. We don't need it anymore because the internal input has the matching type that will prevent submitting the value when unchecked: https://codepen.io/jcfranco/pen/XWBwmoy?editors=1001